sl_functionality.cc

```diff
@@ -200,6 +263,41 @@ void ServiceLayerFunctionality::broadcast_chirp(const std::string& username, Chi
       client_->put(follower_key, monitor_serial);
     }
   }
+
+  std::string message = chirp.text();
+  size_t pos = message.find("#");
+  while (pos != std::string::npos) {
+    message = message.substr(pos);
+    std::string hashtag = message;
+    size_t space = hashtag.find(" ");
+    if (space != std::string::npos) {
+      hashtag = hashtag.substr(0,space);
+    }
+    stream_chirp(chirp, hashtag);
+    message = message.substr(1);
+    pos = message.find("#");
+  }
+}
+
+void ServiceLayerFunctionality::stream_chirp(Chirp chirp, const std::string& hashtag) {
+  std::cout<<"Someone is trying to stream a chirp with hashtag "<<hashtag<<std::endl;
+  const std::string kStreamingKey = "streaming::" + hashtag;
+  std::string streaming_serial = client_->get(kStreamingKey);
+  Streamers streamers;
+  streamers.ParseFromString(streaming_serial);
+
+  for(int i = 0; i < streamers.username_size(); i++) {
+    std::string stream_key = "stream::"+streamers.username(i);
```

Comment:
Move hard-coded string to a constant?

Reply:
I forgot to fork the repository instead of just being on Krishna's repo but done on the forked version (new pr to come I am so sorry)